nflow-engine/src/main/java/io/nflow/engine/internal/executor/WorkflowDispatcher.java

```diff
@@ -80,6 +86,8 @@ public void run() {
           sleep(false);
         }
       }
+    } catch (IOException | ReflectiveOperationException e) {
```

Comment:
True, my bad. How about catching (Exception e) there? I can't remove the catch as I can't add a throws to the run() method, and adding a surrounding try catch around the definition search is kind of silly, as I do want to shutdown stuff in the finally block after that and not go into the loop.